docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -0,0 +1,768 @@
+
+Resulting property and field values:
+
+|Property |Value  |Notes  |
+|---------|---------|---------|
+| Date    | 8/1/2019 12:00:00 AM -07:00||
+| TemperatureC| 0 | Not set because property name matching is case-sensitive.|
+| Summary| Hot||
+| SummaryField| null | Not set because it's a field, not a property.|
+| DatesAvailable | 8/1/2019 12:00:00 AM -07:00<br>8/2/2019 12:00:00 AM -07:00 | `IList<T>` and `IEnumerable<T>` implementations are supported. |
+| TemperatureRanges | Cold, 20 High -10 Low<br>Hot, 60 High 20 Low| `Dictionary<string,TValue>` is supported. |
+| SummaryWords | Cool<br>Windy<br>Humid | Arrays are supported. |
+
+Overloads of <xref:System.Text.Json.JsonSerializer.Deserialize*> let you deserialize from a `Stream`.  Async versions of the `Stream` overloads are available.
+
+### Default deserialization behavior
+
+* If the JSON contains a value for a read-only property, the value is ignored and no exception is thrown.
+* Deserialization to reference types without a parameterless constructor isn't supported.
+* Deserialization to [immutable objects](https://github.com/dotnet/corefx/issues/38569) or [read-only properties](https://github.com/dotnet/corefx/issues/38163) isn't supported.
+* Comments or trailing commas in the JSON trigger exceptions.
+* A nullable property is set to null if the JSON has a null value or the property is missing from the JSON. For information about how to differentiate these two situations, see [GitHub issue 37485](https://github.com/dotnet/corefx/issues/37485).
+
+## Serialize to formatted JSON
+
+Set <xref:System.Text.Json.JsonSerializerOptions.WriteIndented> to true:
+
+```csharp
+var options = new JsonSerializerOptions
+{
+    WriteIndented = true,
+};
+json = JsonSerializer.Serialize(weatherForecast, options);
+```
+
+Example type to be serialized and JSON output:
+
+```csharp
+class WeatherForecast
+{
+    public DateTimeOffset Date { get; set; }
+    public int TemperatureC { get; set; }
+    public string Summary { get; set; }
+}
+```
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot"
+}
+```
+
+## Allow comments and trailing commas
+
+Set <xref:System.Text.Json.JsonSerializerOptions.ReadCommentHandling> to `JsonCommentHandling.Skip`, and set <xref:System.Text.Json.JsonSerializerOptions.AllowTrailingCommas> to true:
+
+```csharp
+var options = new JsonSerializerOptions
+{
+    ReadCommentHandling = JsonCommentHandling.Skip,
+    AllowTrailingCommas = true
+};
+var weatherForecast = JsonSerializer.Deserialize<WeatherForecast>(json);
+```
+
+Example JSON with comments and a trailing comma:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25, // Fahrenheit 77
+  "Summary": "Hot", /* Zharko */
+}
+```
+
+## Customize JSON names
+
+This section explains how to:
+
+* Customize individual property names
+* Convert all property names to camel case
+* Implement a custom property naming policy
+* Convert dictionary keys to camel case
```

Comment:
We should add an example about converting enums to their label names (rather than numbers) using JsonStringEnumConverter.
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonstringenumconverter?view=netcore-3.0